components/clinical/plan-definition-editor/src/index.tsx

```diff
@@ -0,0 +1,240 @@
+    newState && onProblemChange(newState)
+  }
+
+  const handleToggleAllEducationClick = (e: ChangeEvent<HTMLInputElement>) => {
+    const newState = educationActions?.map((p) => ({ problemId: p?.elementId as string, state: e.target.checked }))
+
+    newState && onProblemChange(newState)
+  }
+
+  return (
+    <>
+      <StyledSectionToggle>
+        <div>
+          <ToggleInput
+            checked={allProblemsOn}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => handleToggleAllProblemsClick(e)}
+            id="problem-toggle-all"
+          />
+        </div>
+        <label htmlFor="problem-toggle-all">Patient Problems</label>
+      </StyledSectionToggle>
+      <StyledProblemSection>
+        {problemActions?.map((problem, pidx) => {
+          const goal = planDefinition.goal?.find(
+            (g) => problem?.goalId && g?.elementId?.includes(problem.goalId[0] as string)
+          )
+          return (
+            <StyledListItem key={`problem-${pidx}`}>
+              <StyledProblemToggleRow>
+                <div>
+                  <ToggleInput
+                    checked={selectedProblemIds.includes(problem?.elementId as string)}
+                    onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                      handleSingleActionChange([{ problemId: problem?.elementId as string, state: e.target.checked }])
+                    }
+                    id={`problem-${pidx}`}
+                  />
+                </div>
+                <label htmlFor={`problem-${pidx}`}>{problem?.description}</label>
+              </StyledProblemToggleRow>
+              <div>
+                <StyledTargetRow>
+                  <span>
+                    <BullseyeIcon size="large" />
+                  </span>
+                  <div>{goal?.description?.text}</div>
+                </StyledTargetRow>
+
+                {problem?.action?.map((interventionPlan, ipidx) => (
+                  <StyledTargetRow key={`intervention-plan-${ipidx}`}>
+                    <span />
+                    <div>
+                      <p>{interventionPlan?.description}</p>
+                      <ul>
+                        {interventionPlan?.action?.map((intervention, idx) => (
+                          <li key={`intervention-${idx}`}>{intervention?.description}</li>
+                        ))}
+                      </ul>
+                    </div>
+                  </StyledTargetRow>
+                ))}
+              </div>
+            </StyledListItem>
+          )
+        })}
+      </StyledProblemSection>
+      <StyledSectionToggle>
+        <div>
+          <ToggleInput
+            checked={allEducationOn}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => handleToggleAllEducationClick(e)}
+            id="education-toggle-all"
+          />
+        </div>
+        <label htmlFor="education-toggle-all">Patient Education</label>

```

Comment:
Just thinking aloud but I wonder if this is an opportunity for a re-usable component? Even if just a molecule wrapping the atom. One benefit would be that users of this component can't forget to add the `htmlFor`! :p

Reply:
How would you feel about a linting rule instead? 
[https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md)

Reply:
Perfect! I would feel very positive indeed about that

Reply:
Turns out we have it! Think it was introduced for some of our custom input components (ToggleInput)
![image](https://user-images.githubusercontent.com/22618005/201635527-698d75a2-0ffe-4f03-9f94-389bb1208963.png)